server/sublist.go

```diff
@@ -1128,6 +1129,42 @@ func isValidLiteralSubject(tokens []string) bool {
 	return true
 }
 
+// ValidateDestinationSubject returns nil error if the subject is a valid subject mapping destination subject
+func ValidateDestinationSubject(subject string) error {
+	subjectTokens := strings.Split(subject, tsep)
+	sfwc := false
+	for _, t := range subjectTokens {
+		length := len(t)
+		if length == 0 || sfwc {
+			return ErrBadSubjectMappingDestination
+		}
+
+		if length > 4 && t[0] == '{' && t[1] == '{' && t[length-2] == '}' && t[length-1] == '}' {
+			if !PartitionMappingFunctionRegEx.MatchString(t) && !WildcardMappingFunctionRegEx.MatchString(t) {
+				return ErrUnknownMappingDestinationFunction
+			} else {
+				continue
+			}
+		}
+
+		if length > 1 {
+			if strings.ContainsAny(t, "\t\n\f\r ") {
+				return ErrBadSubjectMappingDestination
+			} else {
+				continue
+			}
+		}
+
+		switch t[0] {
+		case fwc:
+			sfwc = true
+		case ' ', '\t', '\n', '\r', '\f':
```

Comment:
the if and the case statement can combine to 

```
if length == 1 && t[0] == fwc {
   sfwc = true
} else if strings.ContainsAny(t, "\t\n\f\r ") {
   return ErrBadSubjectMappingDestination
}
```

Though yours might be a tad faster since sometimes it wont be calling the more complex `strings.ContainsAny`.  This is certainly more readable and maintainable though

Reply:
```
BenchmarkCompare/ContainsAny
BenchmarkCompare/ContainsAny-16         	162492498	         7.037 ns/op
BenchmarkCompare/Rune_compare
BenchmarkCompare/Rune_compare-16        	1000000000	         0.0000001 ns/op
``` its faster, but is it really going to matter :)

Reply:
That's how the code was and I didn't change it but I agree what you suggest is much more elegant and readable and indeed the difference in performance is not going to matter.